spec/lib/medical_records/user_eligibility/client_spec.rb

```diff
@@ -0,0 +1,120 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require 'medical_records/user_eligibility/client'
+
+describe UserEligibility::Client do
+  describe 'User eligibility operations', :vcr do
+    context 'when user ID and ICN are valid' do
+      before do
```

Comment:
Calling Flipper.enable :mhv_medical_records_new_eligibility_check here will enable it for the entire spec run and could cause issues in other specs that rely on this flipper. Could you stub it instead?

```
    before do
      allow(Flipper).to receive(:enabled?).with(:mhv_medical_records_new_eligibility_check).and_return(true)
    end
```

Also you can add this `before do` before the first context so you don't have to repeat it in each context.

```
describe UserEligibility::Client do
  describe 'User eligibility operations', :vcr do
    let(:user_id) { '10000000' }
    let(:icn) { '1000000000V000000' }
    let(:client) { UserEligibility::Client.new(user_id, icn) }

    before do
      allow(Flipper).to receive(:enabled?).with(:mhv_medical_records_new_eligibility_check).and_return(true)
    end
    
    context 'when user ID is nil' ....
```